app/components/works/license_component.html.erb

```diff
@@ -5,7 +5,7 @@
       <%= form.label :license, class: 'col-sm-2 col-form-label' %>
       <div class="col-sm-5">
         <%= form.select :license,
-                        grouped_options_for_select(License.grouped_options, license),
+                        grouped_options_for_select(License.grouped_options(license), license),
```

Comment:
Pass the license to the method so it knows whether to show the merely displayable licenses as well as the selectable ones. (Don't show them unless there's a legacy license.)